encrypt_test.go

```diff
@@ -1345,6 +1359,238 @@ func testNoWriteMessage(t *testing.T, version Version) {
 	}
 }
 
+func TestEncryptSinglePacketV1(t *testing.T) {
+	sender := boxSecretKey{
+		key: RawBoxKey{0x08},
+	}
+	receivers := []BoxPublicKey{boxPublicKey{key: RawBoxKey{0x1}}}
+
+	plaintext := make([]byte, encryptionBlockSize)
+	ciphertext, err := Seal(Version1(), plaintext, sender, receivers)
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	mps := newMsgpackStream(bytes.NewReader(ciphertext))
+
+	var headerBytes []byte
+	_, err = mps.Read(&headerBytes)
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	var block encryptionBlockV1
+
+	// Payload packet.
+	_, err = mps.Read(&block)
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	// Empty footer payload packet.
+	_, err = mps.Read(&block)
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	// Nothing else.
+	_, err = mps.Read(&block)
+	if err != io.EOF {
+		t.Fatalf("err=%v != io.EOF", err)
+	}
+}
+
+func TestEncryptSinglePacketV2(t *testing.T) {
+	sender := boxSecretKey{
+		key: RawBoxKey{0x08},
+	}
+	receivers := []BoxPublicKey{boxPublicKey{key: RawBoxKey{0x1}}}
+
+	plaintext := make([]byte, encryptionBlockSize)
+	ciphertext, err := Seal(Version2(), plaintext, sender, receivers)
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	mps := newMsgpackStream(bytes.NewReader(ciphertext))
+
+	var headerBytes []byte
+	_, err = mps.Read(&headerBytes)
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	var block encryptionBlockV2
+
+	// Payload packet.
+	_, err = mps.Read(&block)
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	if !block.IsFinal {
+		t.Fatal("IsFinal unexpectedly not set")
+	}
+
+	// Nothing else.
+	_, err = mps.Read(&block)
+	if err != io.EOF {
+		t.Fatalf("err=%v != io.EOF", err)
+	}
+}
+
+func TestEncryptSubsequenceV1(t *testing.T) {
```

Comment:
Added a couple more tests to make sure we were folding the isFinal flag into the hash properly.